tests/test_sdist.py

```diff
@@ -74,14 +74,32 @@ def bar():
 def test_executable_bit(sdist_executable_bit):
     sdist = tarfile.open(sdist_executable_bit, 'r:gz')
 
-    assert set((tar.name, tar.mode) for tar in sdist.getmembers()) == {
-        ('executable_bit-1.0.0/PKG-INFO', 420),
-        # We match the executable bit on everything
-        # but PKG-INFO(we create this ourselves)
-        # Note: File perms are in octal, but Python returns it in int
-        ('executable_bit-1.0.0/example-script.py', int('755', 8)),
-        ('executable_bit-1.0.0/example.c', int('644', 8)),
-        ('executable_bit-1.0.0/executable_module.py', int('755', 8)),
-        ('executable_bit-1.0.0/meson.build', int('644', 8)),
-        ('executable_bit-1.0.0/pyproject.toml', int('644', 8)),
+    expected = {
+        'executable_bit-1.0.0/PKG-INFO': None,
+        'executable_bit-1.0.0/example-script.py': True,
+        'executable_bit-1.0.0/example.c': False,
+        'executable_bit-1.0.0/executable_module.py': True,
+        'executable_bit-1.0.0/meson.build': False,
+        'executable_bit-1.0.0/pyproject.toml': False,
     }
+    assert set(tar.name for tar in sdist.getmembers()) == set(expected.keys())
+
+    def has_execbit(mode):
+        # Note: File perms are in octal, but Python returns it in int
+        # We check multiple modes, because Docker may set group permissions to
+        # match owner permissions
+        modes_execbit = (int('755', 8), int('775', 8))
+        modes_nonexecbit = (int('644', 8), int('664', 8))
```

Comment:
```suggestion
        modes_execbit = 0o755, 0o775
        modes_nonexecbit = 0o644, 0o664
```

Reply:
nicer, thanks

Reply:
CI is not happy because of an unrelated test failure, so let's fix that first before committing this change

Reply:
Yeah, I don't know why I didn't catch it when the code was introduced :stuck_out_tongue: